sources/models/src/modeled_types/kubernetes.rs

```diff
@@ -1002,6 +1006,163 @@ mod test_topology_manager_policy {
 
 // =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=
 
+/// imageGCHighThresholdPercent is the percent of disk usage after which image
+/// garbage collection is always run. The percent is calculated by dividing this
+/// field value by 100, so this field must be between 0 and 100, inclusive. When
+/// specified, the value must be greater than imageGCLowThresholdPercent.
+/// Default: 85
+/// https://kubernetes.io/docs/reference/config-api/kubelet-config.v1beta1/
+
+#[derive(Debug, Clone, Eq, PartialEq, Hash)]
+pub struct ImageGCHighThresholdPercent {
+    // Important note: custom types currently need to have `string_impls_for!()`
+    // after their TryFrom<&str> implementation. (this requires that the inner
+    // type of a custom Kubernetes settings type is a String).
+    // Ideally, `string_impls_for!()` would be accompanied by `i32_impls_for!()`
+    // and macros for other `inner` types.
+    inner: String,
+}
+
+impl TryFrom<&str> for ImageGCHighThresholdPercent {
+    type Error = error::Error;
+
+    fn try_from(input: &str) -> Result<Self, Self::Error> {
+        let parsed_input: i32 = input
+            .parse::<i32>()
+            .context(error::ParseIntSnafu { input })?;
+        ensure!(
+            !input.is_empty(),
+            error::InvalidImageGCHighThresholdPercentSnafu {
+                input,
+                msg: "must not be empty",
+            }
+        );
+        ensure!(
+            (IMAGE_GC_THRESHOLD_MIN..=IMAGE_GC_THRESHOLD_MAX).contains(&parsed_input),
+            error::InvalidImageGCHighThresholdPercentSnafu {
+                input,
+                msg: "must be between 0 and 100 (inclusive)"
+            }
+        );
+
+        Ok(ImageGCHighThresholdPercent {
+            inner: input.to_owned(),
+        })
+    }
+}
+string_impls_for!(ImageGCHighThresholdPercent, "ImageGCHighThresholdPercent");
+
+#[cfg(test)]
+mod test_image_gc_high_threshold_percent {
+    use super::ImageGCHighThresholdPercent;
+    use std::convert::TryFrom;
+
+    // test 1: good values should succeed
+    #[test]
+    fn image_gc_high_threshold_percent_between_0_and_100_inclusive() {
+        for ok in &["0", "1", "99", "100"] {
+            ImageGCHighThresholdPercent::try_from(*ok).unwrap();
+        }
+    }
+
+    // test 2: values too low should return Errors
+    #[test]
+    fn image_gc_high_threshold_percent_less_than_0_fails() {
+        for err in &["-1"] {
+            ImageGCHighThresholdPercent::try_from(*err).unwrap_err();
+        }
```

Comment:
A `for` might be over-killing for a single test case, same for similar test cases